contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -305,10 +320,51 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoTriggerCount(uint256 newAutoTriggerCount) external onlyAdmin {
+        uint256 prevAutoTriggerCount = _autoTriggerCount;
+        if (prevAutoTriggerCount == newAutoTriggerCount) {
+            return;
+        }
+
+        _autoTriggerCount = newAutoTriggerCount;
+
+        emit AutoTriggerCountUpdated({
+            prevAutoTriggerCount: prevAutoTriggerCount,
+            newAutoTriggerCount: newAutoTriggerCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function trigger() external nonReentrant {
+        address[] memory values = _pools.values();
+        uint256 numOfPools = values.length;
+        uint256 startIndex = _nextTriggerIndex;
```

Comment:
no need for startIndex, can use _nextTriggerIndex directly

Reply:
It's being used inside the loop, so better cache it first.

Reply:
oh makes sense